processing/src/main/java/io/druid/segment/QueryableIndexStorageAdapter.java

```diff
@@ -1026,10 +1051,21 @@ public boolean matches()
       }
     }
 
-    // Currently unused, except by unit tests, since filters always support bitmap indexes currently.
-    // This will change when non-String dimensions are added.
     @Override
-    public ValueMatcher makeValueMatcher(String dimension, final Predicate predicate)
+    public ValueMatcher makeValueMatcher(String dimension, final DruidPredicateFactory predicateFactory)
+    {
+      ValueType type = getTypeForDimension(dimension);
+      switch (type) {
+        case LONG:
+          return makeLongValueMatcher(dimension, predicateFactory.makeLongPredicate());
+        case STRING:
+          return makeStringValueMatcher(dimension, predicateFactory.makeStringPredicate());
+        default:
+          throw new UOE("Cannot make ValueMatcher for type[%s]", type);
```

Comment:
This also makes me wonder if ColumnSelectorBitmapIndexSelector should return a null-column-equivalent bitmap index from getBitmapIndex on an unfilterable column. That would mean we can deal with those columns in the preFilter stage rather than postFilter stage, which should be better.